src/core/network-openssl.c

```diff
@@ -701,6 +702,22 @@ GIOChannel *net_connect_ip_ssl(IPADDR *ip, int port, IPADDR *my_ip, SERVER_REC *
 	return ssl_handle;
 }
 
+GIOChannel *net_start_ssl(SERVER_REC *server)
+{
+	GIOChannel *handle, *ssl_handle;
+
+	g_return_val_if_fail(server != NULL, NULL);
+
+	handle = net_sendbuffer_handle(server->handle);
+	if (handle == NULL)
+		return NULL;
+	ssl_handle  = irssi_ssl_get_iochannel(handle, server->connrec->port, server);
+	if (ssl_handle == NULL)
+		g_io_channel_unref(handle);
```

Comment:
Nope, when the call above fails the original handle is left untouched since `g_io_channel_unix_get_fd` is the only function that operates on the `handle` parameter of `irssi_ssl_get_iochannel` before it reaches the function tail where it can't fail anymore.